core/src/test/scala/cats/derived/show.scala

```diff
@@ -0,0 +1,31 @@
+package cats.derived
+import cats.Show
+import cats.derived.TestDefns.{Foo, Node, Tree, Leaf}
+import MkShow._
+
+class ShowTests extends KittensSuite {
+
+  test("for simple product") {
+    import cats.instances.all._
+    val foo = Foo(42, Option("Hello"))
+    assert(Show[Foo].show(foo) == "Foo(i = 42, b = Some(Hello))")
```

Comment:
It's the coproduct instances which are responsible for rendering which branch of the coproduct you're traversing to, so if you start at the product then, as here you'll just get a rendering for the content of the product. Try adding a `Typeable` instance at the product level and use it's `describe` method.